yarpl/include/yarpl/observable/ObservableOperator.h

```diff
@@ -41,11 +41,6 @@ class ObservableOperator : public Observable<D> {
         : observable_(std::move(observable)), observer_(std::move(observer)) {
       assert(observable_);
       assert(observer_);
-
-      // We expect to be heap-allocated; until this subscription finishes (is
-      // canceled; completes; error's out), hold a reference so we are not
-      // deallocated (by the subscriber).
-      Refcounted::incRef(*this);
```

Comment:
Ideally, in a refcounted world, the Reference object saves us from any need to delay destruction. If nobody has a reference to us, we just by definition unreachable. I think this incRef/decRef was put here to be consistent with the other inc/decRefs (but still not needed). 

@lehecka looks good?